### PR TITLE
Inventory cleanup

### PR DESCRIPTION
**Pull request by Bryntet**
## Description
In this PR, I have cleaned up things surrounding the `/clear` and `/give` commands, so that they don't use manual ranges and things like this.

I have also added a bit of error handling for PumpkinError if it occurs inside of a command.

## Testing
I have tested and made sure that both these commands work as intended.

## Checklist
Things need to be done before this Pull Request can be merged.

- [x] Code is well-formatted and adheres to project style guidelines: `cargo fmt`
- [x] Code does not produce any clippy warnings `cargo clippy`

### CODE REVIEW

**Comment by Bryntet** [comment]
Before this gets merged, I'd like @user622628252416 to review the command changes I did (for error handling)

File: pumpkin/src/command/commands/cmd_clear.rs [hunk 1]
```diff
@@ -4,43 +4,62 @@ use async_trait::async_trait;
...
+            ))
+        }
+    } else if item_count == 0 {
+        TextComponent::text_string(format!("No items were found on {target_count} players"))
+            .color_named(NamedColor::Red)
+    } else {
+        TextComponent::text_string(format!(
+            "Removed {item_count} item(s) from {target_count} players"
+        ))
+    }
```
Comment (user622628252416) [inline pumpkin/src/command/commands/cmd_clear.rs:60]: You can use pattern matching with slices to make this a bit cleaner, eg.

```
match targets {
    [target] if item_count = 0 => ...,
    [target] => ...,
    _ => ...
}
```

File: pumpkin/src/command/commands/cmd_clear.rs [hunk 2]
```diff
@@ -101,25 +96,13 @@ impl CommandExecutor for ClearSelfExecutor {
...
-            ))
-            .color_named(NamedColor::Red)
-        } else {
-            TextComponent::text_string(format!(
-                "Removed {} item(s) on player {}",
-                items_count, target.gameprofile.name
-            ))
-        };
+        let hold_target = vec![target];
+        let msg = clear_command_text_output(item_count, &hold_target);
```
Comment (user622628252416) [inline pumpkin/src/command/commands/cmd_clear.rs:106]: Im not sure but could just doing `&[target]` work here instead of allocating a vec?

**Review by user622628252416 — COMMENTED** [review]
The command stuff looks good to me :)

File: pumpkin/src/command/commands/cmd_clear.rs (hunk 2 above)
Comment (Bryntet) [reply pumpkin/src/command/commands/cmd_clear.rs:106]: no, rust yelled at me for doing that at first :P

Comment (user622628252416) [reply pumpkin/src/command/commands/cmd_clear.rs:102]: Well, that's how it shall be then

**Comment by Snowiiii** [comment]
LGTM.
Thanks @Bryntet

**Pull request merged**